mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { __, _x } from '@wordpress/i18n';
+import {
+  SummaryList,
+  SummaryListPlaceholder,
+  SummaryNumber,
+} from '@woocommerce/components/build';
+import { select, useSelect } from '@wordpress/data';
+import { MailPoet } from '../../../../../../mailpoet';
+import { OverviewSection, storeName } from '../../store';
+import { automationHasEmails } from '../../helpers/automation';
+import { locale } from '../../../../../config';
+
+function getEmailPercentage(
+  type: 'opened' | 'clicked',
+  period: 'current' | 'previous' = 'current',
+): number | undefined {
+  const overview = select(storeName).getSection('overview');
+  if (overview.data === undefined) {
+    return undefined;
+  }
+
+  const data = overview.data[type] ?? null;
+  const total = overview.data?.total ?? null;
+  if (!data || !total || !data[period] || !total[period]) {
+    return 0;
+  }
+
+  return (data[period] * 100) / total[period];
+}
+
+function getEmailDelta(type: 'opened' | 'clicked'): number | undefined {
+  const current = getEmailPercentage(type, 'current');
+  const previous = getEmailPercentage(type, 'previous');
+  if (current === undefined || previous === undefined) {
+    return undefined;
+  }
+
+  if (previous === 0) {
+    return 0;
+  }
+
+  const newValue = current > previous ? current - previous : previous - current;
+  return (newValue / previous) * 100;
+}
+
+function getWooCommerceTotal(
+  type: 'revenue' | 'orders',
+  period: 'current' | 'previous' = 'current',
+): number | undefined {
+  const overview = select(storeName).getSection('overview');
+  if (overview.data === undefined) {
+    return undefined;
+  }
+
+  const data = overview.data[type] ?? null;
+  if (!data || !data[period]) {
+    return 0;
+  }
+
+  return data[period] as number;
+}
+
+function getWooCommerceDelta(type: 'revenue' | 'orders'): number | undefined {
+  const current = getWooCommerceTotal(type, 'current');
+  const previous = getWooCommerceTotal(type, 'previous');
+
+  if (current === undefined || previous === undefined) {
+    return undefined;
+  }
+  const newValue = current > previous ? current - previous : previous - current;
+  return (newValue / previous) * 100;
+}
+
+export function Overview(): JSX.Element | null {
+  const { overview } = useSelect((s) => ({
+    overview: s(storeName).getSection('overview'),
+  })) as { overview: OverviewSection };
```

Comment:
Why `as { overview: OverviewSection }`, is the auto-typing not working for your store?

Reply:
`getSection()` returns `Section`, but as those sections have different data structures, they get special types, in this case 

```
export type OverviewSection = Section & {
  data: undefined | OverviewSectionData;
};
```

So the Section type is good to access `id` or `name` or `endpoint`. If you want to access the data you would need to specify the specific Section.

Reply:
Right, I see 👍 There are ways to tell TS to understand based on the params, but I guess not needed now.